changelog.d/7089.bugfix

```diff
@@ -0,0 +1 @@
+Fix typing issue in federation client found with mypy.
```

Comment:
it'd be good to express this in terms of a user-visible change, even if we're not entirely sure what the change is...

> Fix a bug which could cause occasional "Failed to get PDU" errors.

Reply:
I updated this (and added it is for the federation API).